app/assets/javascripts/components/articles/articles_handler.jsx

```diff
@@ -1,37 +1,52 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import createReactClass from 'create-react-class';
+import { connect } from "react-redux";
+
 import ArticleList from './article_list.jsx';
-import UIActions from '../../actions/ui_actions.js';
 import AssignmentList from '../assignments/assignment_list.jsx';
 import ServerActions from '../../actions/server_actions.js';
 import AvailableArticles from '../articles/available_articles.jsx';
 import CourseOresPlot from './course_ores_plot.jsx';
 import CategoryHandler from '../categories/category_handler.jsx';
+import { fetchArticles, sortArticles } from "../../actions/articles_actions.js";
 
 const ArticlesHandler = createReactClass({
   displayName: 'ArticlesHandler',
 
   propTypes: {
     course_id: PropTypes.string,
     current_user: PropTypes.object,
-    course: PropTypes.object
+    course: PropTypes.object,
+    fetchArticles: PropTypes.func,
+    limitReached: PropTypes.bool,
+    limit: PropTypes.number,
+    articles: PropTypes.array
   },
 
   componentWillMount() {
-    ServerActions.fetch('articles', this.props.course_id);
     ServerActions.fetch('assignments', this.props.course_id);
+    return this.props.fetchArticles(this.props.course_id, this.props.limit);
+  },
+
+  showMore() {
+    return this.props.fetchArticles(this.props.course_id, this.props.limit + 100);
```

Comment:
This should go up by 500 each time. Fetching 500 initially and then just 100 more each extra fetch will be pretty tedious for very active programs.